### ISSUE TITLE
[ACRA 2.12.0beta] Issue with deck casing

### ISSUE THREAD

**Issue by david-allison-1** [ankidroid/Anki-Android#6636]
https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/51d27d6e-35a0-44cd-8aa9-c6efbef1c195

```
com.ichi2.anki.exception.ManuallyReportedException: We looked for deck "Derma::Chapter 10: Benign and malignant skin tumours" and instead got deck "Derma::Chapter 10: Benign and malignant skin tumours+".
at com.ichi2.anki.AnkiDroidApp.sendExceptionReport(AnkiDroidApp.java:1)
at com.ichi2.libanki.Decks.byName(Decks.java:5)
at com.ichi2.libanki.Decks.id(Decks.java:7)
at com.ichi2.libanki.Decks.id(Decks.java:2)
at com.ichi2.libanki.Decks.id(Decks.java:1)
at com.ichi2.libanki.Decks._ensureParents(Decks.java:8)
at com.ichi2.libanki.Decks._checkDeckTree(Decks.java:21)
at com.ichi2.libanki.Decks.checkIntegrity(Decks.java:3)
at com.ichi2.libanki.sched.Sched.deckDueList(Sched.java:2)
at com.ichi2.libanki.sched.SchedV2.deckDueTree(SchedV2.java:2)
at com.ichi2.async.CollectionTask.doInBackgroundLoadDeckCounts(CollectionTask.java:3)
at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:43)
at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:1)
at android.os.AsyncTask$2.call(AsyncTask.java:333)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
at java.lang.Thread.run(Thread.java:764)
```

**Comment by david-allison-1** [comment]
@Arthur-Milchior Probably fixed in #6620? If not, we have data for a good test case.

**Comment by Arthur-Milchior** [comment]
I fear that you're wrong. #6620 should not have fixed it. It should fixes NullPointerException and probably nothing else.

In this case, I would have hoped that 09f2c3ac639987e347e12282540cd06842c66fdf would have fixed it. But this was incorporated in 2.12.0beta3 so I guess it's not the correct answer either.

The fact that a "+" was added to the deck name means that two decks with the same name were found and were corrected in `_checkDeckTree`

If the check found any error, the name map is constructed again at the end of the check. The user probably did realize that a "+" was added to a deck's name but appart from that the app theoretically should still have worked.



I guess it shows I was wrong when I thought I could just wait and create the map again at the end. The map is a whole part of the class and is going to be used to correct the tree, so it should be consistent all the time. I can correct that
